cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizerInstantiatorService.java

```diff
@@ -0,0 +1,281 @@
+    }
+  }
+
+  private Properties createExtensionProperties() {
+    Properties extensionProperties = new Properties();
+    for (Map.Entry<String, String> entry :
+      cConf.getValByRegex("^" + Constants.Security.Authorization.AUTHORIZER_EXTENSION_CONFIG_PREFIX).entrySet()) {
+      // remove the prefix
+      String key =
+        entry.getKey().substring(Constants.Security.Authorization.AUTHORIZER_EXTENSION_CONFIG_PREFIX.length());
+      // trim the leading periods
+      if (key.startsWith(".")) {
+        char[] keyChars = key.toCharArray();
+        int i = 0;
+        for (; i < keyChars.length; i++) {
+          if (keyChars[i] != '.') {
+            break;
+          }
+        }
+        key = key.substring(i);
+      }
+      extensionProperties.put(key, entry.getValue());
+    }
+    return extensionProperties;
+  }
+
+  private AuthorizerClassLoader createAuthorizerClassLoader(File authorizerExtensionJar)
+    throws IOException, InvalidAuthorizerException {
+    LOG.info("Creating authorization extension using jar {}.", authorizerExtensionJar);
+    BundleJarUtil.unJar(Locations.toLocation(authorizerExtensionJar), tmpDir);
+    return new AuthorizerClassLoader(tmpDir);
+  }
+
+  @SuppressWarnings("unchecked")
+  private Class<? extends Authorizer> loadAuthorizerClass(File authorizerExtensionJar)
+    throws IOException, InvalidAuthorizerException {
+    String authorizerClassName = getAuthorizerClassName(authorizerExtensionJar);
+    Class<?> authorizerClass;
+    try {
+      authorizerClass = authorizerClassLoader.loadClass(authorizerClassName);
+    } catch (ClassNotFoundException e) {
+      throw new InvalidAuthorizerException(
+        String.format("Authorizer extension class %s not found. Please make sure that the right class is specified " +
+                        "in the extension jar's manifest located at %s.",
+                      authorizerClassName, authorizerExtensionJar), e);
+    }
+    if (!Authorizer.class.isAssignableFrom(authorizerClass)) {
+      throw new InvalidAuthorizerException(
+        String.format("Class %s defined as %s in the authorization extension's manifest at %s must implement %s",
+                      authorizerClass.getName(), Attributes.Name.MAIN_CLASS, authorizerExtensionJar,
+                      Authorizer.class.getName()));
+    }
+    return (Class<? extends Authorizer>) authorizerClass;
+  }
+
+  /**
+   * Inspect the given auth extension jar to find the {@link Authorizer} class contained in it.
+   *
+   * @param authorizerExtensionJar the bundled jar file for the authorizer extension
+   * @return name of the class defined as the {@link Attributes.Name#MAIN_CLASS} in the authorizer extension jar
+   * @throws IOException if there was an exception opening the jar file
+   */
+  private String getAuthorizerClassName(File authorizerExtensionJar) throws IOException, InvalidAuthorizerException {
+    Location authExtensionLocation = Locations.toLocation(authorizerExtensionJar);
```

Comment:
In fact, since you are unjaring the jar anyway, why not just get the manifest as part of the process?